### PR TITLE
detector: fix an issue with deleting all teams/tags/authorized users/teas=ms

### PR DESCRIPTION
**Pull request by dwalasek**
detector: fix an issue with deleting all teams/tags/authorized users/authorized teams

To remove teams/tags/authorized users/authorize from detector resource, the API client must send an empty array. Sending any of these fields missing (null) has no effect

### CODE REVIEW

**Comment by dwalasek** [comment]
I have read the CLA Document and I hereby sign the CLA

File: go.mod [hunk 1]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/hashicorp/go-retryablehttp v0.7.7
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.30.0
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/signalfx/signalfx-go v1.36.0
+	github.com/signalfx/signalfx-go v1.36.1-0.20240613150536-43123c77db5e
```
Comment (atoulme) [inline go.mod]: Why not use 1.37.0?

File: signalfx/resource_signalfx_detector.go [hunk 2]
```diff
@@ -306,7 +306,7 @@ func getPayloadDetector(d *schema.ResourceData) (*detector.CreateUpdateDetectorR
 	maxDelay := int32(d.Get("max_delay").(int) * 1000)
 	minDelay := int32(d.Get("min_delay").(int) * 1000)
 
-	var tags []string
+	tags := []string{}
```
Comment (atoulme) [inline signalfx/resource_signalfx_detector.go:309]: Any chance you can add a test for this?

**Review by bgola-signalfx — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (dwalasek) [reply go.mod]: Fixed it.

File: signalfx/resource_signalfx_detector.go (hunk 2 above)
Comment (dwalasek) [reply signalfx/resource_signalfx_detector.go:309]: Good point. I've added some scenarios to: resource_signalfx_detector_test

**Review by atoulme — APPROVED** [review]

**Pull request merged**